sql/as_of_test.go

```diff
@@ -59,13 +61,25 @@ func TestAsOfTime(t *testing.T) {
 		t.Fatal("unexpected error:", err)
 	}
 
-	if _, err := db.Exec("CREATE DATABASE d"); err != nil {
+	var logicalTS int64
+	if err := db.QueryRow("CREATE DATABASE d; SELECT cluster_logical_timestamp()::int").Scan(&logicalTS); err != nil {
 		t.Fatal("unexpected error:", err)
 	}
-	if err := db.QueryRow("SELECT now()").Scan(&tm); err != nil {
-		t.Fatal(err)
-	}
-	tsDBExists := tm.Format(time.RFC3339Nano)
+	// Make sure we see the DB create by using a TS after its txn time.
```

Comment:
Wait a minute, all nodes in this cluster share a clock, right? It shouldn't be possible to get a transaction back at a timestamp that's ahead of any physical clock in the system (hard to imagine even when clocks jump backward).

Do you have an explanation? Otherwise, I think you're in for a bug hunt!